### ISSUE TITLE
No logic rules were found in ptr_prompt.py.

### ISSUE THREAD

**Issue by 161220012** [thunlp/OpenPrompt#55]
I have read **_PTR: Prompt Tuning with Rules for Text Classification_** and want to find corresponding code implementation in ptr_prompt.py. However, I find that class PTRVerbalizer.process_logits() just simply sums logits of each label word. I want to ask if I should add logical rules by myself to make my own prompt program more aware of entity type information? 
Concretely, as for relation "per:date_of_death": ["person", "was", "died", "on", "date"], I wish more attention should be paid to "person" and "date".

**Comment by Achazwl** [comment]
In [PTR](https://arxiv.org/pdf/2105.11259.pdf), the logic rule used is the `∧` ( logic and ) operation. For your example, `per:date_of_death` is formalized as: subject entity is of type `person` **and** object entity is of type `date` **and** the relation is `was died on`. In probability, p(AB)=p(A) p(B) by assuming conditional independence. In this implementation, log logits are passed to the process_logits function. Therefore, calculating the summation of log logits is an equivalent way to calculate the product of probability.

**Comment by 161220012** [comment]
Thank you for your reply, that helps me a lot.
